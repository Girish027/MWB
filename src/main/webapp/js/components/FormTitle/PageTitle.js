import React from 'react';
import PropTypes from 'prop-types';
import { LegacyGrid, LegacyRow, LegacyColumn } from '@tfs/ui-components';
import ObjectUtils from 'utils/ObjectUtils';

const PageTitle = (props) => {
  const {
    title, description, properties, formContext,
  } = props;
  let hideTitle = false;
  let hideDescription = false;

  if (formContext.rootSettings) {
    hideTitle = ObjectUtils.get(['rootSettings', 'title', 'hidden'], formContext, false);
    hideDescription = ObjectUtils.get(['rootSettings', 'description', 'hidden'], formContext, false);
  }

  const showPageTitle = !ObjectUtils.isEmptyOrNull(title) && !hideTitle;
  const showPageDesc = !ObjectUtils.isEmptyOrNull(description) && !hideDescription;

  return (
    <LegacyGrid>
      { showPageTitle || showPageDesc
        ? (
          <LegacyRow>
            <LegacyColumn size={10}>
              { showPageTitle
              && (
                <div
                  style={{
                    color: '#333333',
                    fontSize: '20px',
                    lineHeight: '30px',
                    fontWeight: 'bold',
                    marginBottom: '4px',
                  }}
                >
                  {title}
                </div>
              )
              }
              { showPageDesc
                && (
                  <div
                    style={{
                      color: '#9B9B9B',
                      fontSize: '15px',
                      lineHeight: '21px',
                      fontWeight: 'normal',
                    }}
                  >
                    {description}
                  </div>
                )
              }
            </LegacyColumn>
          </LegacyRow>
        )
        : null
      }
      <LegacyRow>
        { properties.map((element) => (
          <LegacyColumn
            key={element.name}
            size={formContext[element.name] ? formContext[element.name].size : 12}
          >
            {element.content}
          </LegacyColumn>
        ))}
      </LegacyRow>
    </LegacyGrid>
  );
};

PageTitle.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  properties: PropTypes.array,
  formContext: PropTypes.object,
};

PageTitle.defaultProps = {
  title: '',
  description: '',
  properties: [],
  formContext: {},
};

export default PageTitle;
