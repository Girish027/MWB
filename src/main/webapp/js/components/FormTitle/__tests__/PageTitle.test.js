import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PageTitle from 'components/FormTitle/PageTitle';

describe('<PageTitle />', () => {
  let wrapper;


  const props = {
    formContext: {
      abc: 'asdasd',
    },
    properties: [{
      name: 'abc',
    }, {
      name: 'bcd',
    }],
    title: 'Title',
    description: 'Title Description',
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<PageTitle
    {...propsObj}
  />);

  const getMountWrapper = (propsObj) => mount(<PageTitle
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getMountWrapper();
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly with default props', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
