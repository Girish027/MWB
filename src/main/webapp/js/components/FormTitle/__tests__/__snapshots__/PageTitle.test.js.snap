// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> Snapshots renders correctly with default props 1`] = `
<Deprecated>
  <Deprecated>
    <Deprecated
      size={10}
    >
      <div
        style={
          Object {
            "color": "#333333",
            "fontSize": "20px",
            "fontWeight": "bold",
            "lineHeight": "30px",
            "marginBottom": "4px",
          }
        }
      >
        Title
      </div>
      <div
        style={
          Object {
            "color": "#9B9B9B",
            "fontSize": "15px",
            "fontWeight": "normal",
            "lineHeight": "21px",
          }
        }
      >
        Title Description
      </div>
    </Deprecated>
  </Deprecated>
  <Deprecated>
    <Deprecated
      key="abc"
    />
    <Deprecated
      key="bcd"
      size={12}
    />
  </Deprecated>
</Deprecated>
`;
